src/arrayMethodReduce.js

```diff
@@ -5,7 +5,30 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator;
+    let start;
+
+    switch (initialValue) {
+      case undefined:
+        if (arguments.length === 2) {
+          accumulator = 'undefined';
+          start = 0;
+        } else {
+          accumulator = this[0];
+          start = 1;
+        }
+        break;
+      default:
+        accumulator = initialValue;
+        start = 0;
+        break;
+    }
```

Comment:
good solution, but switch / case is redundant here
let's just simply set some initial values at the start of the function
and then according to the passed argument will change or won't change them